src/main.rs

```diff
@@ -221,12 +295,53 @@ where
     writeln!(file, "source=(\"{}\")", source)?;
     writeln!(file, "sha256sums=(\"{}\")", sha256)?;
     writeln!(file)?;
+    // Include the prepare, build and check steps for non-binary package.
+    if no_bin {
+        writeln!(file, "prepare() {{")?;
+        writeln!(file, "    cd $pkgname-$pkgver")?;
+        writeln!(file, "    export RUSTUP_TOOLCHAIN=stable")?;
+        writeln!(
+            file,
+            "    cargo fetch --locked --target \"$(rustc -vV | sed -n 's/host: //p')\""
+        )?;
+        writeln!(file, "}}")?;
+        writeln!(file)?;
+        writeln!(file, "build() {{")?;
+        writeln!(file, "    cd $pkgname-$pkgver")?;
+        writeln!(file, "    export RUSTUP_TOOLCHAIN=stable")?;
+        writeln!(file, "    export CARGO_TARGET_DIR=target")?;
+        writeln!(file, "    cargo build --frozen --release --all-features")?;
+        writeln!(file, "}}")?;
+        writeln!(file)?;
+        writeln!(file, "check() {{")?;
+        writeln!(file, "    cd $pkgname-$pkgver")?;
+        writeln!(file, "    export RUSTUP_TOOLCHAIN=stable")?;
+        writeln!(file, "    cargo test --frozen --all-features")?;
+        writeln!(file, "}}")?;
+        writeln!(file)?;
+    }
     writeln!(file, "package() {{")?;
-    writeln!(
-        file,
-        "    install -Dm755 {} -t \"$pkgdir/usr/bin\"",
-        config.binary_name()
-    )?;
+    // Install command for binary differs depending on bin/no_bin.
+    match no_bin {
```

Comment:
As mentioned above, here and elsewhere, please use `if` instead of pattern matching on a boolean.

Reply:
actioned